pandas/io/common.py

```diff
@@ -823,6 +852,96 @@ def get_handle(
     )
 
 
+class _BytesTarFile(tarfile.TarFile, BytesIO):
```

Comment:
Is it possible to share code with the _BytesZipFile class? Maybe _BytesCompressMixin from which both classes inherit?

Reply:
Can you comment on why we need this wrapper?

Reply:
I tried extracting some of their code into a Mixin, but found that there's little room for abstraction. Except for the three lines of `write`, none of the methods are identical. Since adding a Mixin also makes the code harder to follow, I'd prefer to keep the duplication. Having said that, if you see a good way of abstracting here, I'm more than open to it!

Reply:
Added a comment in 887fd10f80e380d7bd907894927a8a368556dce1.